src/App.js

```diff
@@ -1,23 +1,77 @@
 import React, { Component } from 'react';
 // import logo from './logo.svg';
 import './App.css';
+import './fontello.css';
 import ProgressBar from './ProgressBar';
+import Categories from './Categories';
+import Tasks from './Tasks';
 
 class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <section className="App-header">
-          <h2>To-Do List</h2>
-        </section>
-        <ProgressBar />
-        <main>
-            {/*<Categories />*/}
-            {/*<Tasks />*/}
-        </main>
-      </div>
-    );
-  }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            list : [
+                {num: 1, name: "Category 1", subcategories: [
+                    {num: 1, name: "Subcategory 1", subcategories: []},
+                    {num: 2, name: "Subcategory 2", subcategories: []}
+                ], tasks: [{name: "To-do item 1"}] },
+                {num: 2, name: "Category 2", subcategories: [
+                    {num: 2, name: "Subcategory 2", subcategories: []}
+                ], tasks: [
+                    {name: "To-do item 1"}
+                ]},
+                {num: 3, name: "Category 3", subcategories: [], tasks: []},
+                {num: 4, name: "Category 4", subcategories: [
+                    {num: 1, name: "Subcategory 1", subcategories: []},
+                    {num: 2, name: "Subcategory 2", subcategories: []},
+                    {num: 3, name: "Subcategory 3", subcategories: []}
+                ], tasks: [
+                    {name: "To-do item 1"},
+                    {name: "To-do item 2"},
+                    {name: "To-do item 3"}
+                ]}
+            ],
+            mode: 0, // 0 - display, 1 - edit
+            activeCategory: null,
+        };
+
+        this.onSelectCategory = this.onSelectCategory.bind(this);
+    }
+
+    onSelectCategory(category) {
```

Comment:
It's a good practice to use `handle` prefix in event handlers names -> `handleSelectCategory`